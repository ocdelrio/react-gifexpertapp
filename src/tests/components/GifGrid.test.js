import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs')

describe('Test de GifGrid', () => {

    const categoria = "Pokemon";
    let wrapper;

    test('Debe de mostrarse correctamente', () => {

        //userFetchGifs mokeado
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        wrapper = shallow(<GifGrid category={categoria}/>)
        expect(wrapper).toMatchSnapshot();
    });    

    test('Debe mostar items cuando se cargan imagenes.', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/test/unitarios/imagen.jpg',
            title: 'ABC'
        }];

        //userFetchGifs mokeado
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        wrapper = shallow(<GifGrid category={categoria}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    });
    


});
