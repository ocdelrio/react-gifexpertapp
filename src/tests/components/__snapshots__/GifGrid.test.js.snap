// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de GifGrid Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Pokemon
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test de GifGrid Debe mostar items cuando se cargan imagenes. 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Pokemon
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="ABC"
      url="https://localhost/test/unitarios/imagen.jpg"
    />
  </div>
</Fragment>
`;
